rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp

```diff
@@ -76,6 +77,88 @@ SqliteWrapper::~SqliteWrapper()
   }
 }
 
+void SqliteWrapper::apply_pragma_settings(
+  std::vector<std::string> & pragmas,
+  rosbag2_storage::storage_interfaces::IOFlag io_flag)
+{
+  // sqlite pragmas are assigned with either '= value' or '(value)' syntax, depending on pragma
+  const std::string pragma_assign = "=";
+  const std::string pragma_bracket = "(";
+
+  // Apply default pragmas if not overridden by user setting
+  {
+    // Used to check whether db is readable
+    const std::string schema = "schema_version";
+
+    typedef std::map<std::string, std::string> pragmas_map;
+    pragmas_map default_pragmas = {
+      {schema, ""}
+    };
+    if (io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_WRITE) {
+      const pragmas_map write_default_pragmas = {
+        {"journal_mode", pragma_assign + "WAL"},
+        {"synchronous", pragma_assign + "NORMAL"}
+      };
+      default_pragmas.insert(write_default_pragmas.begin(), write_default_pragmas.end());
+    }
+    for (const auto & kv : default_pragmas) {
+      auto key = kv.first;
+      auto default_assignment = kv.second;
+      auto pragma_in_settings = std::find_if(
+        pragmas.begin(), pragmas.end(),
+        [key](const auto & pragma) {
+          return pragma.rfind(key, 0) == 0;
+        });
+      if (pragma_in_settings == pragmas.end()) {
+        pragmas.push_back(key + default_assignment);
+      }
+    }
+  }
+
+  for (auto & pragma : pragmas) {
+    if (pragma.empty()) {
+      continue;
+    }
+
+    // Clean the setting from trailing characters
+    const std::string trailing_set("; \t\f\v\n\r");
```

Comment:
I feel like that functionality should be concerning the parser, not the application of the pragmas. I would say at this point, we should assume a syntactical correct pragma setting.